src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java

```diff
@@ -108,16 +107,64 @@ public static boolean isJobTracked(final String jobName) {
     return !DatadogUtilities.isJobBlacklisted(jobName) && DatadogUtilities.isJobWhitelisted(jobName);
   }
 
+  /**
+   * Retrieve the list of tags from the Config file if regex Jobs was checked
+   *  @param jobName - A string containing the name of some job
+   *  @return - A Map of values containing the key and value of each Datadog tag to apply to the metric/event
+   */
+  public static Map<String,String> getRegexJobTags(final String jobName) {
+    Map<String,String> tags = new HashMap<String,String>();
+    final List<List<String>> whitelistRegex = DatadogUtilities.regexJoblistStringtoList( DatadogUtilities.getWhitelist() );
+    // Each jobInfo is a list containing one regex, and a variable number of tags
+    for (List<String> jobInfo: whitelistRegex) {
+      Pattern p = Pattern.compile(jobInfo.get(0));
+      Matcher m = p.matcher(jobName);
+      if(m.matches()) {
+        for(int i = 1; i < jobInfo.size(); i++) {
+          String[] tagItem = jobInfo.get(i).split(":");
+          if(Character.toString(tagItem[1].charAt(0)).equals("$")) {
+            try {
+              tags.put(tagItem[0], m.group(Character.getNumericValue(tagItem[1].charAt(1))));
+            } catch(IndexOutOfBoundsException e) {
+              logger.fine(String.format("Specified a capture group that doesn't exist, not applying tag: " + Arrays.toString(tagItem) + " " + e));
```

Comment:
Could use `StringBuffer` here

Reply:
There should be a non trivial performance bump by changing all of the `String.format` to something like StringBuilder or StringBuffer. We should change all occurrences in a separate PR though as it seems a little out of scope here.

Reply:
ok but then either you use `format` or `+` sign concatenation but both doesn't make sense to me.